src/dragonball/src/device_manager/legacy.rs

```diff
@@ -86,14 +90,22 @@ pub(crate) mod x86_64 {
         pub fn create_manager(bus: &mut IoManager, vm_fd: Option<Arc<VmFd>>) -> Result<Self> {
             let (com1_device, com1_eventfd) =
                 Self::create_com_device(bus, vm_fd.as_ref(), COM1_IRQ, COM1_PORT1)?;
+            METRICS.write().unwrap().serial.insert(
+                String::from(COM1_NAME),
+                com1_device.lock().unwrap().metrics(),
+            );
             let (com2_device, com2_eventfd) =
                 Self::create_com_device(bus, vm_fd.as_ref(), COM2_IRQ, COM2_PORT1)?;
+            METRICS.write().unwrap().serial.insert(
+                String::from(COM2_NAME),
+                com2_device.lock().unwrap().metrics(),
+            );
 
             let exit_evt = EventFd::new(libc::EFD_NONBLOCK).map_err(Error::EventFd)?;
-            let i8042_device = Arc::new(Mutex::new(I8042Device::new(
-                EventFdTrigger::new(exit_evt.try_clone().map_err(Error::EventFd)?),
-                Arc::new(I8042DeviceMetrics::default()),
-            )));
+            let i8042_device = Arc::new(Mutex::new(I8042Device::new(EventFdTrigger::new(
+                exit_evt.try_clone().map_err(Error::EventFd)?,
+            ))));
+            METRICS.write().unwrap().i8042 = i8042_device.lock().unwrap().metrics();
```

Comment:
seem like there is no registeration for `rtc_pl031` device?

Reply:
The `rtc_pl031` device is only registered under the `aarch64`. It works on line 198.